File: 1781. Sum of Beauty of All Substrings/Solution.py
class Solution(object):
    def beautySum(self, s):
        """
        :type s: str
        :rtype: int
        """
        sum = 0

        for i in range(len(s)):
            map = {}
            for j in range(i,len(s)):
                map[s[j]] = map.get(s[j],0)+1

                maxf = max(map.values())
                minf = min(map.values())

                sum+=maxf-minf

        return sum


        

File: 1781. Sum of Beauty of All Substrings/Solution.java
class Solution {
    public int beautySum(String s) {
        int n = s.length();
        int sum = 0;

        for(int i=0;i<n;i++){
            HashMap<Character,Integer> map = new HashMap<>();
            for(int j = i;j<n;j++){
                map.put(s.charAt(j),map.getOrDefault(s.charAt(j),0)+1);

                int max = Integer.MIN_VALUE;
                int min = Integer.MAX_VALUE;

                for(int v:map.values()){
                    max = Math.max(max,v);
                    min = Math.min(min,v);
                }
                sum+=max-min;
            }
        }
        return sum;
    }
}